### DISCUSSION TITLE
My R2 cdn bucket is not working in microfeed.

### DISCUSSION THREAD

**Discussion by caston1981** [microfeed/microfeed#110] [General]
I have installed Microfeed on Cloudfare OK.
I will just give you my domain and URL as it takes forever to blank everything out on all my screenshots. 
It's https://computer-repairs-near-me.com/

Here are the details of my cdn bucket in cloudflare:

![285119566-02576b52-d73b-4a7f-9562-66bc9f1a1030](https://github.com/microfeed/microfeed/assets/40859915/aff3be6d-e541-4f0d-a53a-0921653d9704)

Here is what it looks like in the Microfeed settings:

![Screenshot from 2023-11-23 15-00-45](https://github.com/microfeed/microfeed/assets/40859915/1a0158d1-f647-4330-bebb-71611be7bc01)

Here is what happens if I try to post an image:
![Screenshot from 2023-11-23 15-03-45](https://github.com/microfeed/microfeed/assets/40859915/57198cd8-d48f-450f-b24a-789808da329d)

Here is the link to the resulting post:

https://computer-repairs-near-me.com/i/test-444-wIlQBrhCyDa/

Any suggestions?

**Comment by wenbinf** [comment]
To troubleshoot , here are some questions that might help -

- Are you able to see the file in your r2 bucket via cloudflare dashboard? 

- did you set r2 api key correctly? eg, grant the correct permission to allow uploading files 

my best guess is that the file is not uploaded successfully due to invalid r2 api key.

**Reply by caston1981** [reply]
It doesn't show in the dashboard so it appears to have not uploaded.

I set the R2_Access_Key_ID and R2_Secret_Access_Key when following the install instructions. I still have them saved on my computer if I need to check they work.
And I did grant the access permission under "settings" as shown:
![285125945-ee9775b3-0068-4b18-88b4-4262c788accd](https://github.com/microfeed/microfeed/assets/40859915/50c13785-77ea-473e-b208-cc799b46a19b)

**Reply by wenbinf** [reply]
you need to grant r2 api key permission on the page where you get the r2 api key. 

one quick test to see if those api key works is to run curl + s3 compatible api to upload files.

**Comment by caston1981** [comment]
![Screenshot from 2023-11-23 16-00-51](https://github.com/microfeed/microfeed/assets/40859915/3903e93e-ac9c-482f-80f9-faaa7b0a4169)

I  tried:

curl -X PUT https://cdn.computer-repairs-near-me.com/test1.png \
-H "Authorization: Bearer [api_key]" \
-H "Content-Type: image/png" \
-T ~/test1.png

I used my API keys I had saved and even tried creating new ones but I still get unauthorized. 

I am thinking I should look for some very basic R2 bucket tutorial and try that.

**Reply by wenbinf** [reply]
You should use S3 compatible api to test those r2 api keys: https://developers.cloudflare.com/r2/api/s3/api/

Another thing you can try is to open your browser's developer console, and upload an image/audio/files via microfeed admin, and see if there are any errors in the developer console. It's also possible that some browser extensions might interfere with microfeed's javascript code and makes file uploads fail.

**Comment by caston1981** [comment]
![Screenshot from 2023-11-23 18-34-03](https://github.com/microfeed/microfeed/assets/40859915/6fe34ddb-df56-4bb9-a221-2f25b00425e8)

I tried replicating that folder structure and rename the file to that name then uploading it to the R2 bucket (called cdn) through the cloudflare dashboard and it does show it. https://computer-repairs-near-me.com/i/test-test-SoXRvfcrCXh/

![Screenshot from 2023-11-23 18-54-06](https://github.com/microfeed/microfeed/assets/40859915/c225eaf5-be4d-438f-bbfd-79182ece415f)


I also have another R2 bucket as shown:
![Screenshot from 2023-11-23 18-55-30](https://github.com/microfeed/microfeed/assets/40859915/c2caa2d7-1a85-4464-80da-98618088738d)

This R2 bucket is called "computer-repairs-near-me-com". Perhaps all the files are being uploaded here but it is trying to display them from the other bucket?

**Comment by wenbinf** [comment]
Did you manually create a bucket named "cdn"? If so, please delete it.

You should set public domain name for the bucket automatically created by microfeed (i.g., computer-repairsnear-me-com).

**Reply by caston1981** [reply]
I have tried changing to the other bucket but it doesn't work. At this point I would prefer to just reinstall and ensure that it uses the cdn bucket.

**Comment by caston1981** [comment]
OK it's all working now.

https://www.computer-help-near-me.com/i/test-Hiv7oPLFu38/
Thank you for your help and sorry about the confusion. I was trying to make a cdn R2 bucket rather than just give the project bucket a custom cdn subdomain.
